src/utils.cpp

```diff
@@ -195,6 +199,128 @@ bool is_lit(const Term & l, const Sort & boolsort)
   return false;
 }
 
+void cnf_to_dimacs(Term cnf, std::ostringstream & y)
+{
+  if (cnf->is_value() && cnf->to_string() == "true")
+  {  // empty cnf formula
+    y << "p cnf 0 0\n";
+    return;
+  }
+  TermVec vecs({ cnf });
+  TermVec vecs2;
+  // This while loop separate the clauses, vecs2 will contain all clauses
+  // because every smt::and op will be eliminated. This happens because until
+  // smt::and op is not detected that term is added back to vecs, and as no term
+  // with smt::or as the primOp is touched all clauses shall be separated and be
+  // intact
+  while (!vecs.empty())
+  {
+    Term t = vecs.back();
+    vecs.pop_back();
+    smt::Op op = t->get_op();
+    if (op.prim_op == smt::And)
+    {
+      for (auto u : t)
+      {
+        vecs.push_back(u);
+      }
+    }
+    else
+    {
+      vecs2.push_back(t);
+    }
```

Comment:
Seems like you are assuming that the only operators are `and` and `or`.
This is good, but please assert it. You should probably also add assertions about the fact that the literals are either Boolean variables or their negations (generally speaking, literals can also be complex terms like `x=5+y`).

Reply:
I'll do that